com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/LaunchRequestHandler.java

```diff
@@ -166,7 +176,7 @@
             ListeningConnector listenConnector = connectors.get(0);
             Map<String, Connector.Argument> args = listenConnector.defaultArguments();
             ((Connector.IntegerArgument) args.get("timeout")).setValue(ACCEPT_TIMEOUT);
-            String address = listenConnector.startListening(args);
+            String address = context.getLaunchMode() == LaunchMode.DEBUG ? listenConnector.startListening(args) : null;
 
```

Comment:
add a method isDebugMode in context

Reply:
indeed

Reply:
Revert to avoid confusion caused by the name "DebugMode"